rpc/core/env.go

```diff
@@ -129,3 +129,31 @@ func validateSkipCount(page, perPage int) int {
 
 	return skipCount
 }
+
+func getHeight(currentHeight int64, heightPtr *int64) (int64, error) {
+	if heightPtr != nil {
+		height := *heightPtr
+		if height <= 0 {
+			return 0, fmt.Errorf("height must be greater than 0, but got %d", height)
+		}
+		if height > currentHeight {
+			return 0, fmt.Errorf("height %d must be less than or equal to the current blockchain height %d",
+				height, currentHeight)
+		}
+		currentBase := env.BlockStore.Base()
```

Comment:
Why are we passing currentHeight via param but getting currentBase from environment? Shouldn't both be passed via the same mechanism?

Reply:
because `currentHeight` differs - it can be latestUncommittedHeight or latest block height. current base stays the same